crates/pkgcraft-tools/src/subcmds/pkg/source.rs

```diff
@@ -383,16 +381,23 @@ fn cumulative(limit: u32, targets: PkgTargets, cmd: &Command) -> anyhow::Result<
         let pkgs = targets.clone().ebuild_raw_pkgs();
         progress.reset();
 
-        for result in pkgs.par_map(func).jobs(cmd.jobs) {
-            progress.inc(1);
-            match result {
-                Ok(duration) => cpu_time += duration,
-                Err(e) => {
-                    failed = true;
-                    progress.suspend(|| {
-                        error!("{e}");
-                    });
-                }
+        let result: pkgcraft::Result<Vec<_>> = pkgs
```

Comment:
As this can be run across the entire repo, I'd rather not collect results into a vector.

Reply:
this can be converted to a try_fold then.